tensorflow/python/kernel_tests/self_adjoint_eig_op_test.py

```diff
@@ -150,13 +150,14 @@ def Test(self):
 
 
 if __name__ == '__main__':
-  for dtype in np.float32, np.float64:
+  for dtype in np.float32, np.float64, np.complex64, np.complex128:
     for size in 1, 2, 5, 10:
       for batch_dims in [(), (3,)] + [(3, 2)] * (max(size, size) < 10):
         shape = batch_dims + (size, size)
         name = '%s_%s' % (dtype.__name__, '_'.join(map(str, shape)))
         setattr(SelfAdjointEigTest, 'testSelfAdjointEig_' + name,
                 _GetSelfAdjointEigTest(dtype, shape))
-        setattr(SelfAdjointEigGradTest, 'testSelfAdjointEigGrad_' + name,
-                _GetSelfAdjointEigGradTest(dtype, shape))
+        if dtype is [np.float32, np.float64]:
```

Comment:
Yes, that should fix the gradient.
It might be worth discussing if the gradient should be hermitian instead of the lower triangular matrix, since the user provides a hermitian matrix as the input and the fact that only the lower triangular is used is an implementation detail of the OP. E.g. if the input was a variable it would not be hermitian anymore after applying the gradient.